test/Microsoft.Azure.Devices.Edge.Test.Common/linux/EdgeDaemon.cs

```diff
@@ -94,37 +85,56 @@ static async Task InstallAsync(Option<string> packagesPath, CancellationToken to
                 properties);
         }
 
-        async Task ConfigureAsync(string deviceConnectionString, Option<Uri> proxy, CancellationToken token)
+        public Task ConfigureAsync(Func<DaemonConfiguration, Task<(string, object[])>> config, CancellationToken token)
         {
-            string hostname = (await File.ReadAllTextAsync("/proc/sys/kernel/hostname", token)).Trim();
-            IotHubConnectionStringBuilder builder = IotHubConnectionStringBuilder.Create(deviceConnectionString);
-
-            var properties = new object[] { hostname, builder.DeviceId };
-            string message = "Configured edge daemon for device '{Device}' registered as '{Id}'";
-            proxy.ForEach(
-                p =>
-                {
-                    message += " with proxy '{ProxyUri}'";
-                    properties = properties.Concat(new object[] { p }).ToArray();
-                });
+            var properties = new List<object>();
+            var message = new StringBuilder("Configured edge daemon");
 
-            await Profiler.Run(
+            return Profiler.Run(
                 async () =>
                 {
                     await this.InternalStopAsync(token);
 
                     var yaml = new DaemonConfiguration();
-                    yaml.SetDeviceConnectionString(deviceConnectionString);
-                    yaml.SetDeviceHostname(hostname);
-                    proxy.ForEach(p => yaml.AddHttpsProxy(p));
-                    yaml.Update();
+                    (string m, object[] p) = await config(yaml);
+
+                    message.Append($" {m}");
```

Comment:
It may look weird if variable "m" has multiple lines.

Reply:
```suggestion
                    message.Append($"{m}");
```

Reply:
The value of `m` is being appended to an existing message, so we don't want to remove the space.